packages/graphql-typescript-definitions/src/print/schema/index.ts

```diff
@@ -35,13 +39,34 @@ export function printSchema(schema: GraphQLSchema, options: Options = {}) {
       continue;
     }
 
-    fileBody.push(
-      t.exportNamedDeclaration(tsTypeForRootInputType(type, options), []),
-    );
+    if (isEnumType(type)) {
+      const enumType = tsEnumForType(type, options);
+      fileMap.set(
+        `${enumType.id.name}.ts`,
+        generate(
+          t.file(t.program([t.exportNamedDeclaration(enumType, [])]), [], []),
+        ).code,
+      );
+      fileBody.unshift(
```

Comment:
I find it a bit hard to read that half of these blocks sets the AST, the other half sets a fully rendered page. Can we just set the AST in both cases, and then do a single pass to "stringify" the contents afterwards?

Reply:
the primary purpose of this function is to generate code for type exports, so the file name map is the core state being computed. But we need to keep track of intermediate statements that are not enum type exports so that we can produce the _left over_ types at the end. I'm not sure if there is a better way to do this. We could certainly keep track of the enum `Statement` instances instead, but then we would have to re-iterate over everything again to produce each code fragment.